src/main/java/frc/robot/subsystems/BatterySubsystem.java

```diff
@@ -0,0 +1,199 @@
+  }
+
+  public double getVoltage() {
+    return powerDistribution.getVoltage();
+  }
+
+  public double getInputCurrent() {
+    if (Robot.isSimulation()) {
+      return powerDistribution.getCurrent(1);
+    } else {
+      return powerDistribution.getTotalCurrent();
+    }
+  }
+
+  // If the current is below a certain level, stops the high current timer
+  public void checkCurrent() {
+    if (getInputCurrent() < Constants.highBatteryCurrentThreshold) {
+      stopHCTimer();
+    } else {
+      startHCTimer();
+    }
+  }
+
+  // Displays alerts based on the state of the timer
+  public boolean checkTimer() {
+    if (currentTimer.hasElapsed(Constants.timeInSecondsHighCurrentRed)) {
+      DriverStation.reportWarning("Change the Battery Now! (HCTR)", false);
+      return true;
+    } else if (timer.hasElapsed(Constants.timeInSecondsGeneralRed)) {
+      DriverStation.reportWarning("Change the Battery Now! (GTR)", false);
+      return true;
+    } else if (currentTimer.hasElapsed(Constants.timeInSecondsHighCurrentYellow)) {
+      DriverStation.reportWarning("Change the Battery Soon! (HCTY)", false);
+      return false;
+    } else if (timer.hasElapsed(Constants.timeInSecondsGeneralYellow)) {
+      DriverStation.reportWarning("Change the Battery Soon! (GTY)", false);
+      return false;
+    }
+    return false;
+  }
+
+  public double getGeneralTimer() {
+    Preferences.setDouble("Battery General Timer", timer.get() + storedGeneralTime);
+    return timer.get() + storedGeneralTime;
+  }
+
+  public double getHighCurrentTimer() {
+    Preferences.setDouble("Battery High Current Timer", currentTimer.get() + storedHighCurrentTime);
+    return currentTimer.get() + storedHighCurrentTime;
+  }
+
+  private void startHCTimer() {
+    currentTimer.start();
+  }
+
+  private void stopHCTimer() {
+    currentTimer.stop();
+  }
+
+  public void checkVoltage() {
+    if (getVoltage() < Constants.minVoltageRed) {
+      DriverStation.reportWarning("Change the Battery Now! (VR)", false);
+    } else if (getVoltage() < Constants.minVoltageYellow) {
+      DriverStation.reportWarning("Change the Battery Soon! (VY)", false);
+    } else {
+    }
+  }
+
+  public boolean checkRedVoltage() {
+    return getVoltage() > Constants.minVoltageRed;
+  }
+
+  public void resetTimers() {
+    // Add a key to store timer values if there isn't one already
+    if (!Preferences.containsKey("Battery General Timer")) {
+      Preferences.setDouble("Battery General Timer", 0.0);
+      Preferences.setDouble("Battery High Current Timer", 0.0);
+    } else {
+      storedGeneralTime = Preferences.getDouble("Battery General Timer", 0.0);
+      storedHighCurrentTime = Preferences.getDouble("Battery High Current Timer", 0.0);
+    }
+    timer.reset();
+    currentTimer.reset();
+  }
+
+  // TODO: Find a way to make these not reset every redeploy. Possibly make them save their values
+  // to a file, then
+  // retrieve that file when the robot starts.
```

Comment:
Looks like you've solved this already. Delete comment

Reply:
done